templates/cmd/main.go.tpl

```diff
@@ -22,6 +22,8 @@ import (
 	"github.com/getoutreach/gobox/pkg/events"
 	"github.com/getoutreach/gobox/pkg/trace"
 	"github.com/getoutreach/tollmon/pkg/tollgate"
+	"github.com/getoutreach/stencil-golang/pkg/serviceactivities/shutdown"
+	"github.com/getoutreach/stencil-golang/pkg/serviceactivities/gomaxprocs"
```

Comment:
We def don't want to generate it, since we're trying to reduce the amount of generated code. That being said, the service activity is tied to the implementation here, which is why I decided making it a code dependency was a better idea. Since it's so tied to the version of stencil-golang.

What's your general aversion to keeping it here?

Reply:
I don't love the idea of stencil modules serving double duty. In this case we are conflating template generation and library code. For example what if we want to fix one of these library functions. Does that mean that they also need to re-stencil? To me that loses most of the benefit of keeping this as library level code.

I can see the argument for this case working as is, since we're including a fairly small amount of code. But it's a pattern that could be easily abused.

Reply:
+1 to @artemchernyak-outreach's concern. Mixing templates and libraries feels like an anti-pattern. It seems like templates should depend on a common library (`gobox`?) which provides these helpers.

Reply:
Snippet from my DM to @artemchernyak-outreach:

> The code here is fundamentally the same as if it was being generated by stencil, or put in it’s own module, just as a library inside of stencil-golang. This is meant to be tied to the stencil-golang version, much like the template version if it existed, because it’s tied to the template versions. That’s my main concern with throwing it out into gobox is we force it to be even tighter coupled to gobox and potentially get issues where people can’t upgrade gobox without also re-stenciling (which is a huge antipattern)

That's my main reasoning for keeping this tied to `stencil-golang`. I'm not trying to throw actual libraries into here, these service activities only exist _for_ `stencil-golang` and that's their only use/API compatibility guarantee. We ended up agreeing offline, but happy to chat more if you want @jkinkead

Reply:
Perhaps an easier way to think about it is that this "library" is really just a way to ensure the code isn't generated into the user's repository (as it has no blocks, or things for them to change). Sorta like a VFS, but in a way Go understands it.

Reply:
So what you're saying is that it's effectively a generated file, but pulled out of their source repo to avoid PR churn? I can see the value in that.

Reply:
Yep, that's exactly it. It's tied to the stencil-golang version for that reason 🚀

Reply:
(it's also easier to edit because it's not a go-template, it's just straight go code)